### PR TITLE
 feat: add version toggle and update layout 

### PR DESCRIPTION
**Pull request by azanbinzahid**
## Description
This PR fixes broken header links and adds v2 design into container. No change is required in Header alignment. 

## Linked Ticket
[PROD-2512](https://openedx.atlassian.net/browse/PROD-2512)
[PROD-2403](https://openedx.atlassian.net/browse/PROD-2403)
[PROD-2508](https://openedx.atlassian.net/browse/PROD-2508)

## Screenshot
image
image

### CODE REVIEW

**Comment by azanbinzahid** [comment]
Ignore the coverage drop here. Header unit tests do not exist at this moment.

**Review by DawoudSheraz — APPROVED** [review]

**Review by Ali-D-Akbar — APPROVED** [review]

File: src/SupportToolsTab/constants.js [hunk 1]
```diff
@@ -2,6 +2,6 @@ export const LEARNER_INFO_TAB = 'learner-information';
 export const FEATURE_BASED_ENROLLMENT_TAB = 'feature-based-enrollment';
 
 export const TAB_PATH_MAP = {
-  [LEARNER_INFO_TAB]: '/v2/learner_information',
+  [LEARNER_INFO_TAB]: '/v2/users',
```
Comment (DawoudSheraz) [inline src/SupportToolsTab/constants.js]: why did we change this back to `/users`? We used learner_information to be consistent with how the data is present with new view.

File: src/supportHeader/ToggleVersion.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    if (window.location.href.indexOf('/v2') !== -1) {
+      if (window.location.href.indexOf('/v2/feature_based_enrollments') !== -1) {
+        alert('Cannot Switch to Old Design for this tab');
+      } else {
+        window.location.href = window.location.href.replace('/v2', '');
+      }
+    } else {
+      window.location.href = window.location.href.replace(config.BASE_URL, `${config.BASE_URL}/v2`);
+    }
+  };
```
Comment (DawoudSheraz) [inline src/supportHeader/ToggleVersion.jsx]: lets use frontend-platform history for replacement and react-router location for location checks for the sake of consistency. react-router location, which will be accessible via props, makes it easier to compare paths and make decisions.

File: src/supportHeader/ToggleVersion.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <Form.Check
```
Comment (DawoudSheraz) [inline src/supportHeader/ToggleVersion.jsx:24]: We should add some text before the switch indicating old/new view etc.

File: src/supportHeader/ToggleVersion.jsx (hunk 2 above)
Comment (azanbinzahid) [reply src/supportHeader/ToggleVersion.jsx]: since `Header` component is not under `Routes` that's why we are unable to access location props.

File: src/SupportToolsTab/constants.js (hunk 1 above)
Comment (azanbinzahid) [reply src/SupportToolsTab/constants.js]: I changed them to reduce checks on toggling but now I have changed as you suggested.

File: src/supportHeader/ToggleVersion.jsx (hunk 2 above)
Comment (DawoudSheraz) [reply src/supportHeader/ToggleVersion.jsx]: Oh, right.

**Review by ansabgillani — APPROVED** [review]

File: src/supportHeader/ToggleVersion.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/supportHeader/ToggleVersion.jsx:24]: Yes, a message or a hover text might be good here.

File: src/SupportToolsTab/constants.js [hunk 4]
```diff
@@ -2,6 +2,6 @@ export const LEARNER_INFO_TAB = 'learner-information';
 export const FEATURE_BASED_ENROLLMENT_TAB = 'feature-based-enrollment';
 
 export const TAB_PATH_MAP = {
-  [LEARNER_INFO_TAB]: '/v2/learner_information',
+  [LEARNER_INFO_TAB]: '/v2/learner_infromation',
```
Comment (DawoudSheraz) [inline src/SupportToolsTab/constants.js]: typo: information

File: src/supportHeader/ToggleVersion.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React, { useContext } from 'react';
```
Comment (DawoudSheraz) [inline src/supportHeader/ToggleVersion.jsx:1]: we can skip this file from coverage but this is a temporary file.

Comment (azanbinzahid) [reply src/supportHeader/ToggleVersion.jsx:1]: how do we skip file? is there some config?

File: src/supportHeader/ToggleVersion.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      window.location.href = window.location.href.replace('/users', TAB_PATH_MAP['learner-information']);
+    } else {
+      window.location.href = window.location.href.replace(config.BASE_URL, `${config.BASE_URL}/v2`);
+    }
+  };
+  return (
+    <Form.Check
+      type="switch"
+      id="custom-switch"
+      label={<div style={{ marginTop: '4px', fontSize: 'smaller' }}>V2</div>}
```
Comment (DawoudSheraz) [inline src/supportHeader/ToggleVersion.jsx]: nit: reword this to New Experience or something like New Experience Activated. V2 does not deliver a complete meaning.

Comment (azanbinzahid) [reply src/supportHeader/ToggleVersion.jsx]: wouldn't it be too long? or is it okay?

File: src/supportHeader/ToggleVersion.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      window.location.href = window.location.href.replace('/users', TAB_PATH_MAP['learner-information']);
+    } else {
+      window.location.href = window.location.href.replace(config.BASE_URL, `${config.BASE_URL}/v2`);
+    }
+  };
+  return (
+    <Form.Check
+      type="switch"
+      id="custom-switch"
+      label={<div style={{ marginTop: '4px', fontSize: '1vmax' }}>New Experiance</div>}
```
Comment (DawoudSheraz) [inline src/supportHeader/ToggleVersion.jsx]: nit:
```suggestion
      label={<div style={{ marginTop: '4px', fontSize: '1vmax' }}>New Experience</div>}
```

**Review by DawoudSheraz — APPROVED** [review]
A nit that I have added as a suggestion.
